src/com/skillsup/dao/UserInFileDaoImpl.java

```diff
@@ -0,0 +1,77 @@
+package com.skillsup.dao;
+
+import com.skillsup.model.Gender;
+import com.skillsup.model.User;
+import com.skillsup.model.UserRole;
+
+import java.io.*;
+import java.math.BigDecimal;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+
+public class UserInFileDaoImpl implements UserDAO {
+    private static final String PATH_USER_DB_FILE = "E:\\onlineShop\\src\\com\\skillsup\\resources\\userDB.txt";
+    private static final User ADMIN = new User("admin", "admin", UserRole.ADMIN);
+
+    @Override
+    public void saveUser(User user) {
+        try {
+            FileWriter fileWriter = new FileWriter(PATH_USER_DB_FILE, true);
+            String userString = user.getUsername() + "!" + user.getPassword() + "!" +
```

Comment:
"!" should be extracted as const -> final String SEPARATOR = "!" or smth like that

Reply:
And You are absolutly right again! That would be more clever!